clang/lib/DPCT/AnalysisInfo.cpp

```diff
@@ -2831,6 +2847,101 @@ std::shared_ptr<MemVarInfo> MemVarInfo::buildMemVarInfo(const VarDecl *Var) {
   }
   return DpctGlobalInfo::getInstance().insertMemVarInfo(Var);
 }
+
+void MemVarInfo::migrateWithDeviceGlobal(const VarDecl *MemVar) {
+  auto &SM = DpctGlobalInfo::getSourceManager();
+  auto &Ctx = DpctGlobalInfo::getContext();
+  auto &MacroArgMap = DpctGlobalInfo::getMacroArgRecordMap();
+  auto TSI = MemVar->getTypeSourceInfo();
+  auto OriginTL = TSI->getTypeLoc();
+  auto TL = OriginTL;
+  auto BegLoc = MemVar->getBeginLoc();
+  if (BegLoc.isMacroID()) {
+    BegLoc = SM.getExpansionLoc(BegLoc);
+  }
+  auto LocInfo = DpctGlobalInfo::getLocInfo(BegLoc);
+  std::string Dims;
+  bool IsArray = OriginTL.getType()->isArrayType();
+  // 1.Remove bracket after var name
+  while (auto ATL = TL.getAs<clang::ArrayTypeLoc>()) {
+    auto BRange = ATL.getBracketsRange();
+    auto RT =
+        ReplaceText(SM.getSpellingLoc(BRange.getBegin()),
+                    SM.getSpellingLoc(BRange.getEnd()).getLocWithOffset(1), "");
+    DpctGlobalInfo::getInstance().addReplacement(RT.getReplacement(Ctx));
+    Dims += "[";
+    std::string SizeStr;
+    if (clang::Expr *SE = ATL.getSizeExpr()) {
+      auto SizeLoc = SE->getBeginLoc();
+      if (SM.isMacroArgExpansion(SizeLoc)) {
+        auto Iter =
+            MacroArgMap.find(getCombinedStrFromLoc(SM.getSpellingLoc(SizeLoc)));
```

Comment:
What if the array size contains more tokens e.g. [size+5]

Reply:
I think this issue can wait to fixed when we encounter the real case. Because it's complicated to analyze each part of expression.